apps/guardian-ui/src/components/SetConfiguration.tsx

```diff
@@ -85,6 +90,10 @@ export const SetConfiguration: React.FC<Props> = ({ next }: Props) => {
         );
         setNetwork(walletModule.consensus?.network.toString() || network);
         setBitcoinRpc(walletModule.local?.bitcoin_rpc || bitcoinRpc);
+        setClientDefaultBitcoinRpc(
+          walletModule.consensus?.client_default_bitcoin_rpc ||
+            clientDefaultBitcoinRpc
+        );
```

Comment:
Hmmm this is a bit of an anti-pattern but not your fault. But these are using variables that are not being included in the `useEffect` dependencies, so they may be stale references. Typically you'd just wrap this in a condition and not call the `set*` method, rather than re-passing its existing state:
```suggestion
        if (walletModule.consensus?.client_default_bitcoin_rpc) {
          setClientDefaultBitcoinRpc(walletModule.consensus.client_default_bitcoin_rpc);
        }
```
However, `walletModule` isn't in the `useEffect` dependencies either so it's kind of a wash. This whole `useEffect` probably needs to be re-thought.